contracts/src/token/erc20/utils/safe_erc20.rs

```diff
@@ -15,15 +15,16 @@ pub use sol::*;
 use stylus_sdk::{
     call::{MethodError, RawCall},
     contract::address,
-    evm::gas_left,
     function_selector,
     prelude::storage,
     storage::TopLevelStorage,
     stylus_proc::{public, SolidityError},
     types::AddressVM,
 };
 
-use crate::token::erc20;
+use crate::{
+    token::erc20, utils::reentrant_call_handler::ReentrantCallHandler,
```

Comment:
IMHO would be cool to have just `utils::ReentrantCallHandler`, so I suggest re-exporting from `reentrant_call_handler`.